src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/AddressServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.cozyhome.onlineshop.userservice.security.service.impl;
+
+import com.cozyhome.onlineshop.dto.user.AddressIdDto;
+import com.cozyhome.onlineshop.dto.user.AddressRequest;
+import com.cozyhome.onlineshop.dto.user.AddressResponse;
+import com.cozyhome.onlineshop.exception.DataAlreadyExistException;
+import com.cozyhome.onlineshop.exception.DataNotFoundException;
+import com.cozyhome.onlineshop.userservice.model.Address;
+import com.cozyhome.onlineshop.userservice.model.User;
+import com.cozyhome.onlineshop.userservice.repository.AddressRepository;
+import com.cozyhome.onlineshop.userservice.repository.UserRepository;
+import com.cozyhome.onlineshop.userservice.security.service.AddressService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class AddressServiceImpl implements AddressService {
+
+    private final AddressRepository addressRepository;
+    private final UserRepository userRepository;
+    private final ModelMapper modelMapper;
+    @Override
+    public AddressResponse saveAddress(AddressRequest addressRequest, String userId) {
+        List<Address> addresses = new ArrayList<>();
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new DataNotFoundException(String.format("User with id = %s not found.", userId)));
+
+        Address address = Address.builder()
```

Comment:
propose to use:"Address address = modelMapper.map(addressRequest, Address.class);" instead of 36-55 lines.
![image](https://github.com/NatalliaBulyha/cozy-home-monolith/assets/79281909/e4eb06d2-80f1-4544-b65a-1e2837ca8dee)

Reply:
done